test/integration/mime_type_test.rb

```diff
@@ -162,6 +174,24 @@ def call(params)
     end
 
     it 'sets "Content-Type" header according to "Accept"' do
+      response = @app.get('/custom_from_accept', 'HTTP_ACCEPT' => 'application/custom')
+      response.headers['Content-Type'].must_equal 'application/custom; charset=utf-8'
+      response.body.must_equal                    'custom'
+    end
+
+    it 'sets "Content-Type" header according to "Accept"' do
+      response = @app.get('/custom_from_accept', 'HTTP_ACCEPT' => 'application/custom;q=0.9, application/json;q=0.5')
```

Comment:
I've noticed that there does not seem to be any enforced constraint on line size in Hanami but it does improve readability when lines do not exceed 80 columns (or 90, or 100 – 80 just seems to be the inherited standard from the physical terminal days).